app/assets/javascripts/discourse/initializers/topic-footer-buttons.js.es6

```diff
@@ -0,0 +1,116 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+
+export default {
+  name: "topic-footer-actions",
+
+  initialize() {
+    withPluginApi("0.8", api => {
+      api.registerTopicFooterButton({
+        id: "share",
+        icon: "link",
+        priority: 999,
+        label: "topic.share.title",
+        title: "topic.share.help",
+        action: context => {
```

Comment:
I think this is one of those cases where applying `this` makes more sense. Something like:

```
action() {
  this.appEvents(...);
}
```
I think it's nicer and clearer than `context`

Reply:
indeed, I thought being explicit would be nice, but maybe I could just rely on this being what one would expect.